File: mvvmlibrary/src/main/java/com/longshihan/mvvmlibrary/bus/event/UIChangeLiveData.java
package com.longshihan.mvvmlibrary.bus.event;

import android.arch.lifecycle.LifecycleOwner;
import android.arch.lifecycle.Observer;

import java.util.Map;

public class UIChangeLiveData extends SingleLiveEvent {
    private SingleLiveEvent<String> showDialogEvent;
    private SingleLiveEvent<Void> dismissDialogEvent;
    private SingleLiveEvent<Map<String, Object>> startActivityEvent;
    private SingleLiveEvent<Map<String, Object>> startContainerActivityEvent;
    private SingleLiveEvent<Void> finishEvent;
    private SingleLiveEvent<Void> onBackPressedEvent;

    public SingleLiveEvent<String> getShowDialogEvent() {
        return showDialogEvent = createLiveData(showDialogEvent);
    }

    public SingleLiveEvent<Void> getDismissDialogEvent() {
        return dismissDialogEvent = createLiveData(dismissDialogEvent);
    }

    public SingleLiveEvent<Map<String, Object>> getStartActivityEvent() {
        return startActivityEvent = createLiveData(startActivityEvent);
    }

    public SingleLiveEvent<Map<String, Object>> getStartContainerActivityEvent() {
        return startContainerActivityEvent = createLiveData(startContainerActivityEvent);
    }

    public SingleLiveEvent<Void> getFinishEvent() {
        return finishEvent = createLiveData(finishEvent);
    }

    public SingleLiveEvent<Void> getOnBackPressedEvent() {
        return onBackPressedEvent = createLiveData(onBackPressedEvent);
    }

    private SingleLiveEvent createLiveData(SingleLiveEvent liveData) {
        if (liveData == null) {
            liveData = new SingleLiveEvent();
        }
        return liveData;
    }

    @Override
    public void observe(LifecycleOwner owner, Observer observer) {
        super.observe(owner, observer);
    }
}

File: mvvmlibrary/src/main/java/com/longshihan/mvvmlibrary/bus/event/UIChangeLiveData.kt
/*
package com.longshihan.mvvmlibrary.bus.event

*/
/**
 * Created by LONGHE001.
 * @time 2019/7/24 0024
 * @des
 * @function
 *//*

class UIChangeLiveData : SingleLiveEvent() {
    private var showDialogEvent: SingleLiveEvent<String>? = null
    private var dismissDialogEvent: SingleLiveEvent<Void>? = null
    private var startActivityEvent: SingleLiveEvent<Map<String, Any>>? = null
    private var startContainerActivityEvent: SingleLiveEvent<Map<String, Any>>? = null
    private var finishEvent: SingleLiveEvent<Void>? = null
    private var onBackPressedEvent: SingleLiveEvent<Void>? = null

    fun getShowDialogEvent(): SingleLiveEvent<String> {
        showDialogEvent = createLiveData(showDialogEvent)
        return showDialogEvent!!
    }

    fun getDismissDialogEvent(): SingleLiveEvent<Void> {
        dismissDialogEvent = createLiveData(dismissDialogEvent)
        return dismissDialogEvent!!
    }

    fun getStartActivityEvent(): SingleLiveEvent<Map<String, Any>> {
        startActivityEvent = createLiveData(startActivityEvent)
        return startActivityEvent!!
    }

    fun getStartContainerActivityEvent(): SingleLiveEvent<Map<String, Any>> {
        startContainerActivityEvent = createLiveData(startContainerActivityEvent)
        return startContainerActivityEvent!!
    }

    fun getFinishEvent(): SingleLiveEvent<Void> {
        finishEvent = createLiveData(finishEvent)
        return finishEvent!!
    }

    fun getOnBackPressedEvent(): SingleLiveEvent<Void> {
        onBackPressedEvent = createLiveData(onBackPressedEvent)
        return onBackPressedEvent!!
    }

    private fun createLiveData(liveData: SingleLiveEvent?): SingleLiveEvent {
        var liveData = liveData
        if (liveData == null) {
            liveData = SingleLiveEvent()
        }
        return liveData
    }


}*/
